usr/share/rear/conf/default.conf

```diff
@@ -129,9 +129,18 @@ OS_VERSION=none
 # Useful to inspect the ReaR recovery system content in $TMPDIR/rear.XXXXXXXXXXXXXXX/rootfs/
 # directly without the need to extract it from the initramfs/initrd in the ISO image.
 # KEEP_BUILD_DIR is automatically set to true in debug mode (-d) and in debugscript mode (-D)
-# and by build/default/990_verify_rootfs.sh when it detects errors in the ReaR recovery system.
+# and by build/default/990_verify_rootfs.sh according to KEEP_BUILD_DIR_ON_ERRORS (below)
+# when it detects errors in the ReaR recovery system
 KEEP_BUILD_DIR=""
 
+# Keep the build area on errors (ternary).
+# Set to "y", "Yes", etc, to obtain the old behaviour where KEEP_BUILD_DIR was always set
+# to true to keep the build area when errors in the ReaR recovery system were detected. Set to
+# "n", "No", etc. to never enable KEEP_BUILD_DIR on errors.
+# An empty value KEEP_BUILD_DIR_ON_ERRORS="" is equivalent to 'yes' if running interactively
+# (in a terminal) and 'no' if not (to avoid cluttering /tmp by cron jobs).
+KEEP_BUILD_DIR_ON_ERRORS=""
```

Comment:
Offhandedly I think a new KEEP_BUILD_DIR_ON_ERRORS is not needed
because one can set KEEP_BUILD_DIR="yes" to enforce that.

Furthermore "cluttering /tmp by cron jobs" never was and is no issue in ReaR upstream
because ReaR upstream never provided a cron file.

The cron files are only generated during RPM package build time
at Fedora and Red Hat but I never did that for SUSE / openSUSE, see
https://github.com/rear/rear/issues/1892#issuecomment-410973078

See also the current
https://github.com/rear/rear/issues/2139
and its predecessor
https://github.com/rear/rear/issues/1892
where the reasonig behind is explained why meanwhile the cron stuff is removed,
cf. http://relax-and-recover.org/documentation/release-notes-2-5
that reads (excerpt)
```
In packaging/rpm/rear.spec ...
removed /etc/cron.d/rear and related things (issues #2135 #1855 #1856 #1908 #1892)
```

Reply:
> Offhandedly I think a new KEEP_BUILD_DIR_ON_ERRORS is not needed
> because one can set KEEP_BUILD_DIR="yes" to enforce that.

I thoughts so as well, and this was my first iteration of the PR, but @xjezda00 pointed out that once one sets this, build dir will be kept for successful runs as well - there would be no way to enforce the previous behaviour, which was in some sense reasonable - build dir kept for debugging on errors, removed on success.

> Furthermore "cluttering /tmp by cron jobs" never was and is no issue in ReaR upstream
> because ReaR upstream never provided a cron file.
> 
> The cron files are only generated during RPM package build time
> at Fedora and Red Hat but I never did that for SUSE / openSUSE, see
> [#1892 (comment)](https://github.com/rear/rear/issues/1892#issuecomment-410973078)
> 
> See also the current
> #2139
> and its predecessor
> #1892
> where the reasonig behind is explained why meanwhile the cron stuff is removed,
> cf. http://relax-and-recover.org/documentation/release-notes-2-5
> that reads (excerpt)
> 
> ```
> In packaging/rpm/rear.spec ...
> removed /etc/cron.d/rear and related things (issues #2135 #1855 #1856 #1908 #1892)
> ```

Sorry, I have not noticed this discussion. It may seem that when the cron file is not there, this is less of an issue, but the issue will reappear with any other automation that is either provided by the ReaR upstream (such as the one proposed in #2139) or created by users themselves.

Reply:
I would expect that a user who has manually and intentionally set up
any kind of automated run of whatever program, would monitor the results,
in particular at least check the exit code and/or other kind of errors.

In general - as far as I remember the FHS - programs are free to store
whatever they like in /tmp/ and it is the system admin's task to monitor
the used disk space in /tmp/ and do some cleanup as needed, cf.
https://en.opensuse.org/openSUSE:Tmp_on_tmpfs#.2Ftmp.2F_versus_.2Fvar.2Ftmp.2F